pywhat/identifier.py

```diff
@@ -1,36 +1,48 @@
-import os.path
-
-from pywhat.magic_numbers import FileSignatures
-from pywhat.nameThatHash import Nth
-from pywhat.regex_identifier import RegexIdentifier
-
-
-class Identifier:
-    def __init__(self):
-        self.regex_id = RegexIdentifier()
-        self.file_sig = FileSignatures()
-        self.name_that_hash = Nth()
-
-    def identify(self, text: str, api=False) -> dict:
-        identify_obj = {}
-
-        magic_numbers = None
-        if not api and self.file_exists(text):
-            magic_numbers = self.file_sig.open_binary_scan_magic_nums(text)
-            text = self.file_sig.open_file_loc(text)
-            identify_obj["File Signatures"] = magic_numbers
-        else:
-            text = [text]
-
-        if not magic_numbers:
-            # If file doesn't exist, check to see if the inputted text is
-            # a file in hex format
-            identify_obj["File Signatures"] = self.file_sig.check_magic_nums(text)
-        identify_obj["Regexes"] = self.regex_id.check(text)
-        # get_hashes takes a list of hashes, we split to give it a list
-        # identify_obj["Hashes"] = self.name_that_hash.get_hashes(text.split())
-
-        return identify_obj
-
-    def file_exists(self, text):
-        return os.path.isfile(text)
+import os.path
+import glob
+
+from pywhat.magic_numbers import FileSignatures
+from pywhat.nameThatHash import Nth
+from pywhat.regex_identifier import RegexIdentifier
+
+
+class Identifier:
+    def __init__(self):
+        self.regex_id = RegexIdentifier()
+        self.file_sig = FileSignatures()
+        self.name_that_hash = Nth()
+
+    def identify(self, text: str, api=False) -> dict:
+        identify_obj = {}
+        magic_numbers = {}
+        file_text = {}
+
+        def get_file_data(filepath):
+            short_name = os.path.basename(filepath)
+            magic_numbers[short_name] = self.file_sig.open_binary_scan_magic_nums(filepath)
+            file_text[short_name] = self.file_sig.open_file_loc(filepath)
+
+            # if file doesn't exist, check to see if the inputted text is
+            # a file in hex format
+            if not magic_numbers[short_name]:
+                magic_numbers[short_name] = self.file_sig.check_magic_nums(filepath)
```

Comment:
Functions in functions are not very pythonic, I suggest changing this :-) 

Functions should do one thing and only one thing. Try to follow that principle, if a function does multiple things it should be split into multiple functions. And the name of the function should be what it's doing.

Read this for more info:
https://www.amazon.co.uk/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882